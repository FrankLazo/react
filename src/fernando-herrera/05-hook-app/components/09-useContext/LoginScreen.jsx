import { useContext } from 'react';
import UserContext from './UserContext';

const LoginScreen = () =>
{
	const { setUser } = useContext(UserContext);

	return (
		<div className="min-height-full content has-background-dark p-6 has-text-light">
			<h1 className="has-text-light">Login</h1>
			<hr className="full-rule has-background-grey-dark" />

			<button
				className="button is-primary"
				onClick={ () => setUser({ id: 123, name: 'Frank' }) }
				>
				Login
			</button>
		</div>
	);
};

export default LoginScreen;
