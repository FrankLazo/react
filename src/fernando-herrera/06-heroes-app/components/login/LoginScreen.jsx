import { useContext } from 'react';
import AuthContext from '../../auth/AuthContext';
import { types } from '../../types/types';

const LoginScreen = ({ history }) =>
{
	const { dispatch } = useContext(AuthContext);
	const toPath = localStorage.getItem('lastPath') || '/';

	const handleLogin = () =>
	{
		// history.push('/');
		// history.replace('/');

		dispatch({
			type: types.login,
			payload: {
				user: 'Frank'
			},
		});

		history.replace(toPath);
	};

	return (
		<div className="min-height-full content has-background-light p-4">
			<h1 className="has-text-primary">Login</h1>
			<hr className="has-background-grey-lighter" />

			<button
				className="button is-primary"
				onClick={ handleLogin }
				>
				Login
			</button>
		</div>
	);
};

export default LoginScreen;
