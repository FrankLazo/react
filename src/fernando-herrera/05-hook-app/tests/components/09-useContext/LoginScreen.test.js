import { mount } from 'enzyme';
import LoginScreen from '../../../components/09-useContext/LoginScreen';
import UserContext from '../../../components/09-useContext/UserContext';

describe('<LoginScreen /> tests', () => {
	const setUser = jest.fn();

	const wrapper = mount(
		<UserContext.Provider value={{ setUser }}>
			<LoginScreen />
		</UserContext.Provider>
	);

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should call setUser with the right arguments', () => {
		const button = wrapper.find('button');
		button.simulate('click');

		// Otra manera
		// wrapper.find('button').prop('onClick')();

		expect(setUser).toHaveBeenCalledWith({ id: 123, name: 'Frank' });
	});
});
