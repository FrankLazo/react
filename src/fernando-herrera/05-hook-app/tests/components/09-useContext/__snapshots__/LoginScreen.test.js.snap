// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> tests should show the right component 1`] = `
<div
  className="min-height-full content has-background-dark p-6 has-text-light"
>
  <h1
    className="has-text-light"
  >
    Login
  </h1>
  <hr
    className="full-rule has-background-grey-dark"
  />
  <button
    className="button is-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
